src/main/java/core/basesyntax/FileWork.java

```diff
@@ -1,7 +1,37 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.Arrays;
+
 public class FileWork {
+    public static final String WORDS_SPLIT = "\\W+";
+
     public String[] readFromFile(String fileName) {
-        return null;
+        StringBuilder stringBuilder = new StringBuilder();
+        try {
+            BufferedReader bufferedReader = new BufferedReader(new FileReader(fileName));
+            String line = bufferedReader.readLine();
+            while (line != null) {
+                String[] words = line.toLowerCase().split(WORDS_SPLIT);
+                for (String word : words) {
+                    if (word.startsWith("w") || word.startsWith("W")) {
```

Comment:
```suggestion
                    if (word.startsWith(TARGET_FIRST_LETTER)) {
                    // TARGET_FIRST_LETTER = "w";
```
you don't need to compare uppercase since you called toLowerCase() on the whole line before spliting

Reply:
add